### PR TITLE
Eclipse 2018 12 compatibility

### PR DESCRIPTION
**Pull request by k-oliver**
Version 1.1.0 to be used in Eclipse 2018-12 and older.

### CODE REVIEW

File: eclipse-templatevariables-feature/feature.xml [hunk 1]
```diff
@@ -14,68 +14,68 @@
    </copyright>
 
    <license>
-                                 Apache License
+      Apache License
```
Comment (hohwille) [inline eclipse-templatevariables-feature/feature.xml]: any reason why this shall not be centered, while the lines below remain centered?
looks odd to me...

File: eclipse-templatevariables-feature/feature.xml [hunk 2]
```diff
@@ -14,68 +14,68 @@
...
                            Version 2.0, January 2004
                         http://www.apache.org/licenses/
 
    TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
 
    1. Definitions.
 
-      "License" shall mean the terms and conditions for use, reproduction,
+      &quot;License&quot; shall mean the terms and conditions for use, reproduction,
```
Comment (hohwille) [inline eclipse-templatevariables-feature/feature.xml:25]: was that automatically changed by your XML formatter?

File: eclipse-templatevariables-feature/feature.xml [hunk 3]
```diff
@@ -194,34 +194,39 @@
...
       <update label="TemplateVariables Update Site" url="http://m-m-m.github.io/eclipse-templatevariables/"/>
       <discovery label="mmm TempalteVariables" url="http://m-m-m.github.io/eclipse-templatevariables/"/>
    </url>
 
-   <plugin id="net.sf.mmm.eclipse.templatevariables.plugin" version="0.0.0" unpack="false"/>
+   <plugin
+         id="net.sf.mmm.eclipse.templatevariables.plugin"
+         download-size="0"
+         install-size="0"
+         version="1.1.0.qualifier"
```
Comment (hohwille) [inline eclipse-templatevariables-feature/feature.xml:229]: fine. So `0.0.0` was more a hack.
A perfect fix would be to use `${project.version}.qualifier` in combination with maven filtering.

File: eclipse-templatevariables-plugin/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -2,14 +2,14 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: net.sf.mmm.eclipse.templatevariables.plugin
 Bundle-SymbolicName: net.sf.mmm.eclipse.templatevariables.plugin;singleton:=true
-Bundle-Version: 1.0.0.qualifier
+Bundle-Version: 1.1.0.qualifier
```
Comment (hohwille) [inline eclipse-templatevariables-plugin/META-INF/MANIFEST.MF:5]: same here...

File: eclipse-templatevariables-plugin/src/main/java/net/sf/mmm/eclipse/templatevariables/AbstractMavenTemplateVariableResolver.java [hunk 5]
```diff
@@ -66,13 +69,26 @@ private String getMavenProperty(IProject project) {
...
 
-    return getMavenProperty(((CodeTemplateContext) context).getJavaProject().getProject());
-  }
+        // Workaround for Eclipse versions <4.10
+        try {
+            final Method method = context.getClass().getMethod("getJavaProject");
+            final Object result = method.invoke(context);
+            final IJavaProject javaProject = (IJavaProject)result;
+            return this.getMavenProperty(javaProject.getProject());
+        } catch (final NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException
```
Comment (hohwille) [inline eclipse-templatevariables-plugin/src/main/java/net/sf/mmm/eclipse/templatevariables/AbstractMavenTemplateVariableResolver.java:88]: Seems as this block above is the important part of the fix. Thanks.

**Review by hohwille — CHANGES_REQUESTED** [review]
@k-oliver thanks for this PR. I did not change anything here for a long time as I am too busy with other OSS projects...

Some comments on your changes:
* I personally consider it as an anti-pattern to commit eclipse config files into vcs/git
* I was not even aware that I myself committed `.project` and `.classpath` files to git
* seems I was missing a `.gitignore` so thanks for adding one
* IMHO the `.gitignore` should contain `.*` and we should remove all eclipse config files
* FYI: The diffs are larger than necessary as you are using different formatter settings (e.g. indent by 4 instead of 2 spaces). If you are using Eclipse, you might love devon-ide (https://github.com/devonfw/devon-ide/, successor of https://github.com/oasp/oasp4j-ide). The settings can be found here: https://github.com/m-m-m/settings.

Thanks again for your PR. I could also take over and do the suggested changes. Please let me know if you want to update your fork or want to hand-over to me from here.

**Comment by k-oliver** [comment]
@hohwille I will update my fork....

File: eclipse-templatevariables-feature/feature.xml (hunk 2 above)
Comment (hohwille) [reply eclipse-templatevariables-feature/feature.xml:25]: I do not care much, but just wanted to understand if it is wrong to use regular quotes here (according to my XML understanding this is fine and `&quot;` is only required in double quoted attribute values.)

File: eclipse-templatevariables-feature/feature.xml (hunk 3 above)
Comment (hohwille) [reply eclipse-templatevariables-feature/feature.xml:229]: You do not need to do this change. I can take it over. You already improved/fixed this versioning and it is fine for the upcoming release `1.1.0` so no need to change for this PR.

File: eclipse-templatevariables-feature/feature.xml (hunk 2 above)
Comment (k-oliver) [reply eclipse-templatevariables-feature/feature.xml:25]: It was automatically changed by eclipse. Following https://stackoverflow.com/questions/150423/whats-the-accepted-way-of-storing-quoted-data-in-xml BOTH ways are correct. I would suggest to used ``&quot``. Let me know...

File: eclipse-templatevariables-feature/feature.xml (hunk 1 above)
Comment (k-oliver) [reply eclipse-templatevariables-feature/feature.xml]: Automatically formatted. So no reason.

**Comment by k-oliver** [comment]
@hohwille please review

File: eclipse-templatevariables-feature/feature.xml (hunk 2 above)
Comment (hohwille) [reply eclipse-templatevariables-feature/feature.xml:25]: Sure. Fine this way. Just wanted to understand this...

**Comment by hohwille** [comment]
Awesome. Looks great now. @k-oliver 👍 thank you so much.
I will merge it but I also want to test it once. Might take some more days, but nothing more expected to be done from your end. Once merged I will also push our a new release.

**Comment by hohwille** [comment]
I cloned your fork repo and switched to your feature branch. When I run the maven build I do get this error:
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.22.0:compile (default-compile) on project net.sf.mmm.eclipse.templatevariables.plugin: Compilation failure: Compilation failure:
[ERROR] /projects/mmm/workspaces/main/eclipse-templatevariables-fork/eclipse-templatevariables-plugin/src/main/java/net/sf/mmm/eclipse/templatevariables/AbstractMavenTemplateVariableResolver.java:[79]
[ERROR] 	return this.getMavenProperty(context.getAdapter(IProject.class));
[ERROR] 	                                     ^^^^^^^^^^
[ERROR] The method getAdapter(Class<IProject>) is undefined for the type TemplateContext
[ERROR] 1 problem (1 error)
[ERROR] -> [Help 1]
```

Any ideas? Without this problem being resolved, I can not merge nor release. Did you happen to forget staging some changes in your commit?

**Comment by k-oliver** [comment]
I did not try to build with maven. I was simply using the eclipse export which is building against the eclipse running target platform.
I took a look into it and the solution should be updating the eclipse maven repository. I will update my fork...

**Comment by k-oliver** [comment]
@hohwille I updated my fork. The maven build finished successfully. Please try again.

**Review by hohwille — APPROVED** [review]
@k-oliver Thanks for fixing. The maven build now also works smooth. 👍

**Comment by hohwille** [comment]
@k-oliver do you have any hint regarding #2?
I am unable to build since I merged this PR. Maybe trivial to fix for somebody with huge eclipse p2 experience but I am not sure how to proceed. Seems that dependencies from `MANIFEST.MF` such as `org.eclipse.ui` can not be resolved anymore. My first guess is that the updated repository does not contain this anymore. Need to add an additional repo to pom.xml?

**Pull request merged**